src/DiIiS-NA/D3-GameServer/GSSystem/AISystem/Brains/AggressiveNPCBrain.cs

```diff
@@ -108,9 +109,11 @@ protected virtual int PickPowerToUse()
 		{
 			if (PresetPowers.Count > 0)
```

Comment:
most probably we should filter by `HasImplementationForPowerSNO` and pick random from implemented only powers